package.json

```diff
@@ -3,11 +3,15 @@
   "private": true,
   "scripts": {
     "data": "node scripts/data/index.js",
+    "prebuild": "npm run data",
     "build": "nuxt build",
+    "predev": "npm run data",
     "dev": "nuxt dev",
-    "generate": "nuxt generate",
-    "preview": "nuxt preview",
-    "postinstall": "npm run data && nuxt prepare",
```

Comment:
Moved to pre-scripts since otherwise a new dev would get an error on installing deps since the dato token might not be set yet. Plus netlify.toml doesn't need it anymore.

Reply:
Good! I think that's more conventional anyway.
Would you consider this pattern so we can add more prep scripts?

```json
    "predev": "run-s prep",
    "dev": "...",
    "prebuild": "run-s prep",
    "build": "...",
    "prep": "run-s prep:* --print-label",
    "prep:icons": "svg-sprite-generate -d src/assets/icons -o src/assets/icons-sprite.svg",
    "prep:datocms-types": "graphql-codegen",
    "prep:supabase-types": "...",
    "prep:...": "...",
```

Reply:
I do not fully realise it yet myself, but are you sure it is safe to remove `nuxt prepare`?I suppose the effect would be that I have to build or run the project to have code completion / syntax highlighting in my IDE, right? Also I wonder how it would work with TypeScript check being part of the CI workflow (which we should do). I suggest we either make sure everything is OK with that now, or bring `prepare` back for now and remove it once we do `nuxt typecheck` in CI.

Reply:
@jbmoelker yeah can do, I figured no need for npm-run-all yet so I opted for a pre-script.

@kyrel no strong opinion on this, it's safe to remove but indeed means that you need to run build or dev at least once before types are present.

Reply:
OK, looks like `typecheck` does create the `.nuxt` folder, so all is good